plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaRecordSet.java

```diff
@@ -120,7 +120,7 @@ public RecordCursor cursor()
         private Iterator<ConsumerRecord<byte[], byte[]>> records = emptyIterator();
         private long completedBytes;
 
-        private final FieldValueProvider[] currentRowValues = new FieldValueProvider[columnHandles.size()];
+        private List<FieldValueProvider> currentRowValues = ImmutableList.of();
```

Comment:
any specific reason to move to list from statically defined array?

Reply:
No, only
> - do not copy currentRowValues unnecessarily

Reply:
I don't know if copying is worse or pointer chasing. 🤷 

This is in RecordSet which is most probably performance sensitive. I'll like a second opinion from @Praveen2112 / @raunaqmorarka.

One way to mitigate my concern would be pre-size the ImmutableList instead of empty list.

Reply:
> pointer chasing

It's not obvious how this effect can be obtained/avoided here.
Looks more like legacy code.

> pre-size the ImmutableList

I don't think it matters here.

Reply:
I didn't follow why we changed this code. The existing code was copying object references to a static array. Now we create a new list for each row and then copy object references to it. The array approach seems cheaper, so to me it looks like we should leave it unchanged.
In general if performance is a concern, and the Kafka consumer fetches a batch of records, we should produce pages from the Kafka connector by implementing ConnectorPageSource and more away from the RecordSet api.

Reply:
PageSource is a much larger work - good point that we should consider using it since we fetch around 100k records per batch by default anyway.

In this case I'm just trying to see if we are introducing a regression or not.

Reply:
Would you be open to extracting this change (just the array to list change) to a different PR and continuing there?

A simple timing test on a test which creates a sufficient number of splits (or lower number of messages per split) should be good enough to see if there's visible regression.

Reply:
I've dropped that commit related to cursor refactoring. 
https://github.com/trinodb/trino/pull/14435